svsim/src/main/scala/Workspace.scala

```diff
@@ -0,0 +1,389 @@
+package svsim
+
+import java.io.{BufferedReader, BufferedWriter, File, FileWriter, InputStreamReader}
+import java.nio.file.Paths
+import java.lang.ProcessBuilder.Redirect
+
+case class ModuleInfo(
+  name:  String,
+  ports: Seq[ModuleInfo.Port]) {
+  private[svsim] val instanceName = "dut"
+}
```

Comment:
There will be an API that exposes probes/rwprobes as port-like things. This will be based on the reference types that are now in the FIRRTL 2.0.0 spec: https://github.com/chipsalliance/firrtl-spec/pull/75